diff -U 3 -dHBbrN -- transmission-1.90.orig/libtransmission/session.c transmission-1.90/libtransmission/session.c
--- transmission-1.90.orig/libtransmission/session.c	2010-02-16 19:00:50.000000000 +0300
+++ transmission-1.90/libtransmission/session.c	2010-02-21 10:53:30.637166963 +0300
@@ -241,10 +241,10 @@
 
     tr_bencDictReserve( d, 35 );
     tr_bencDictAddBool( d, TR_PREFS_KEY_BLOCKLIST_ENABLED,        FALSE );
-    tr_bencDictAddBool( d, TR_PREFS_KEY_DHT_ENABLED,              TRUE );
+    tr_bencDictAddBool( d, TR_PREFS_KEY_DHT_ENABLED,              FALSE );
     tr_bencDictAddStr ( d, TR_PREFS_KEY_DOWNLOAD_DIR,             tr_getDefaultDownloadDir( ) );
-    tr_bencDictAddInt ( d, TR_PREFS_KEY_DSPEED,                   100 );
-    tr_bencDictAddBool( d, TR_PREFS_KEY_DSPEED_ENABLED,           FALSE );
+    tr_bencDictAddInt ( d, TR_PREFS_KEY_DSPEED,                   TR_MAX_SPEED_KB );
+    tr_bencDictAddBool( d, TR_PREFS_KEY_DSPEED_ENABLED,           TRUE );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_ENCRYPTION,               TR_DEFAULT_ENCRYPTION );
     tr_bencDictAddStr ( d, TR_PREFS_KEY_INCOMPLETE_DIR,           incompleteDir );
     tr_bencDictAddBool( d, TR_PREFS_KEY_INCOMPLETE_DIR_ENABLED,   FALSE );
@@ -258,8 +258,8 @@
     tr_bencDictAddInt ( d, TR_PREFS_KEY_PEER_PORT_RANDOM_LOW,     49152 );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_PEER_PORT_RANDOM_HIGH,    65535 );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_PEER_SOCKET_TOS,          atoi( TR_DEFAULT_PEER_SOCKET_TOS_STR ) );
-    tr_bencDictAddBool( d, TR_PREFS_KEY_PEX_ENABLED,              TRUE );
-    tr_bencDictAddBool( d, TR_PREFS_KEY_PORT_FORWARDING,          TRUE );
+    tr_bencDictAddBool( d, TR_PREFS_KEY_PEX_ENABLED,              FALSE );
+    tr_bencDictAddBool( d, TR_PREFS_KEY_PORT_FORWARDING,          FALSE );
 #ifdef HAVE_FALLOCATE64
     tr_bencDictAddInt ( d, TR_PREFS_KEY_PREALLOCATION,            TR_PREALLOCATE_FULL );
 #else
@@ -290,8 +290,8 @@
     tr_bencDictAddBool( d, TR_PREFS_KEY_ALT_SPEED_TIME_ENABLED,   FALSE );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_ALT_SPEED_TIME_END,       1020 ); /* 5pm */
     tr_bencDictAddInt ( d, TR_PREFS_KEY_ALT_SPEED_TIME_DAY,       TR_SCHED_ALL );
-    tr_bencDictAddInt ( d, TR_PREFS_KEY_USPEED,                   100 );
-    tr_bencDictAddBool( d, TR_PREFS_KEY_USPEED_ENABLED,           FALSE );
+    tr_bencDictAddInt ( d, TR_PREFS_KEY_USPEED,                   TR_MAX_SPEED_KB );
+    tr_bencDictAddBool( d, TR_PREFS_KEY_USPEED_ENABLED,           TRUE );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_UMASK,                    022 );
     tr_bencDictAddInt ( d, TR_PREFS_KEY_UPLOAD_SLOTS_PER_TORRENT, 14 );
     tr_bencDictAddStr ( d, TR_PREFS_KEY_BIND_ADDRESS_IPV4,        TR_DEFAULT_BIND_ADDRESS_IPV4 );
@@ -1203,6 +1203,7 @@
     assert( tr_isDirection( d ) );
     assert( KB_s >= 0 );
 
+	 if( KB_s > TR_MAX_SPEED_KB ) KB_s = TR_MAX_SPEED_KB;
     s->speedLimit[d] = KB_s;
 
     updateBandwidth( s, d );
@@ -1224,7 +1225,7 @@
     assert( tr_isDirection( d ) );
     assert( tr_isBool( b ) );
 
-    s->speedLimitEnabled[d] = b;
+    s->speedLimitEnabled[d] = TRUE;
 
     updateBandwidth( s, d );
 }
@@ -1249,6 +1250,7 @@
     assert( tr_isDirection( d ) );
     assert( KB_s >= 0 );
 
+	 if( KB_s > TR_MAX_SPEED_KB ) KB_s = TR_MAX_SPEED_KB;
     s->turtle.speedLimit[d] = KB_s;
 
     updateBandwidth( s, d );
@@ -1396,6 +1398,8 @@
 {
     assert( tr_isSession( session ) );
 
+    if( maxGlobalPeers > TR_MAX_PEERS_COUNT ) maxGlobalPeers = TR_MAX_PEERS_COUNT;
+
     tr_fdSetPeerLimit( session, maxGlobalPeers );
 }
 
diff -U 3 -dHBbrN -- transmission-1.90.orig/libtransmission/transmission.h transmission-1.90/libtransmission/transmission.h
--- transmission-1.90.orig/libtransmission/transmission.h	2010-02-16 19:00:50.000000000 +0300
+++ transmission-1.90/libtransmission/transmission.h	2010-02-21 09:49:27.000000000 +0300
@@ -142,6 +142,10 @@
  */
 const char* tr_getDefaultDownloadDir( void );
 
+/* zyxel internal limits */
+#define TR_MAX_SPEED_KB			1024
+#define TR_MAX_PEERS_COUNT		300
+/* zyxel internal limits end */
 
 #define TR_DEFAULT_BIND_ADDRESS_IPV4       "0.0.0.0"
 #define TR_DEFAULT_BIND_ADDRESS_IPV6            "::"
diff -U 3 -dHBbrN -- transmission-1.90.orig/libtransmission/rpcimpl.c transmission-1.90/libtransmission/rpcimpl.c
--- transmission-1.90.orig/libtransmission/rpcimpl.c     2010-02-21 10:45:23.000000000 +0300
+++ transmission-1.90/libtransmission/rpcimpl.c  2010-02-21 10:45:53.369167545 +0300
@@ -1169,10 +1169,10 @@
         tr_sessionUseAltSpeedTime( session, boolVal );
     if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_BLOCKLIST_ENABLED, &boolVal ) )
         tr_blocklistSetEnabled( session, boolVal );
-    if( tr_bencDictFindStr( args_in, TR_PREFS_KEY_DOWNLOAD_DIR, &str ) )
+/*    if( tr_bencDictFindStr( args_in, TR_PREFS_KEY_DOWNLOAD_DIR, &str ) )
         tr_sessionSetDownloadDir( session, str );
     if( tr_bencDictFindStr( args_in, TR_PREFS_KEY_INCOMPLETE_DIR, &str ) )
-        tr_sessionSetIncompleteDir( session, str );
+        tr_sessionSetIncompleteDir( session, str ); */
     if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_INCOMPLETE_DIR_ENABLED, &boolVal ) )
         tr_sessionSetIncompleteDirEnabled( session, boolVal );
     if( tr_bencDictFindInt( args_in, TR_PREFS_KEY_PEER_LIMIT_GLOBAL, &i ) )
@@ -1183,12 +1183,12 @@
         tr_sessionSetPexEnabled( session, boolVal );
     if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_DHT_ENABLED, &boolVal ) )
         tr_sessionSetDHTEnabled( session, boolVal );
-    if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_PEER_PORT_RANDOM_ON_START, &boolVal ) )
+/*    if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_PEER_PORT_RANDOM_ON_START, &boolVal ) )
         tr_sessionSetPeerPortRandomOnStart( session, boolVal );
     if( tr_bencDictFindInt( args_in, TR_PREFS_KEY_PEER_PORT, &i ) )
         tr_sessionSetPeerPort( session, i );
     if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_PORT_FORWARDING, &boolVal ) )
-        tr_sessionSetPortForwardingEnabled( session, boolVal );
+        tr_sessionSetPortForwardingEnabled( session, boolVal ); */
     if( tr_bencDictFindBool( args_in, TR_PREFS_KEY_RENAME_PARTIAL_FILES, &boolVal ) )
         tr_sessionSetIncompleteFileNamingEnabled( session, boolVal );
     if( tr_bencDictFindReal( args_in, "seedRatioLimit", &d ) )
