diff -U 3 -dHBbrN -- busybox-1.15.1.orig/networking/udhcp/leases.c busybox-1.15.1/networking/udhcp/leases.c
--- busybox-1.15.1.orig/networking/udhcp/leases.c	2009-09-12 19:55:36.000000000 +0400
+++ busybox-1.15.1/networking/udhcp/leases.c	2010-09-28 11:10:53.000000000 +0400
@@ -64,13 +64,15 @@
 		oldest->hostname[0] = '\0';
 		if (hostname) {
 			char *p;
-			if (hostname_len > sizeof(oldest->hostname))
-				hostname_len = sizeof(oldest->hostname);
-            		p = safe_strncpy(oldest->hostname, hostname, hostname_len);
+			if (hostname_len >= sizeof(oldest->hostname))
+				hostname_len = sizeof(oldest->hostname) - 1;
+            
+         strncpy(oldest->hostname, hostname, hostname_len);
+         oldest->hostname[hostname_len] = 0;
+         p = oldest->hostname;
 			/* sanitization (s/non-ASCII/^/g) */
 			while (*p) {
-				if (*p < ' ' || *p > 126)
-					*p = '^';
+				if (*p < ' ' || *p > 126) *p = '^';
 				p++;
 			}
 		}
