diff -U 3 -dHBbrN -- transmission-1.83.orig/libtransmission/web.c transmission-1.83/libtransmission/web.c
--- transmission-1.83.orig/libtransmission/web.c	2010-02-10 08:28:03.000000000 +0300
+++ transmission-1.83/libtransmission/web.c	2010-02-10 19:27:00.000000000 +0300
@@ -24,6 +24,7 @@
 #include "utils.h"
 #include "version.h"
 #include "web.h"
+#include <netdb.h>
 
 enum
 {
@@ -455,6 +455,9 @@
     tr_address addr;
     int port = -1;
     char * host = NULL;
+    struct hostent     *he;
+    struct   sockaddr_in   adr;
+     
     struct tr_web_task * task = vtask;
     tr_dns_result lookup_result = TR_DNS_UNTESTED;
 
@@ -474,7 +477,11 @@
         }
         else
         {
-            lookup_result = dns_cache_lookup( task, host, &task->resolved_host );
+        	 if( (he = gethostbyname(host)) ) {
+        		 memcpy(&adr.sin_addr, he->h_addr_list[0], he->h_length);
+        		 task->resolved_host = strdup(inet_ntoa(adr.sin_addr));
+        		 lookup_result = TR_DNS_OK;
+        	 } 
         }
     }
 
